titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/v1/V1SpecPodFactory.java

```diff
@@ -207,6 +208,10 @@ public V1Pod buildV1Pod(Job<?> job, Task task, boolean useKubeScheduler, boolean
         labels.put("v3.job.titus.netflix.com/job-id", job.getId());
         labels.put("v3.job.titus.netflix.com/task-id", taskId);
 
+        // A V1Container has no room to store the original tag that the Image came from, so we store it as an
+        // annotation. Only saving the 'main' one for now.
+        annotations.put(POD_IMAGE_TAG_PREFIX + "main", job.getJobDescriptor().getContainer().getImage().getTag());
```

Comment:
Possibly a dumb question, but jobs can be submitted with a digest instead of a tag, right? Does this behave the right way if a job is submitted with a digest?

Reply:
Great question.

I guess... this would be NPE? 😬 
Or the string `null` ? I'm not super sure.

Reply:
Ok confirmed this is `""` in the case where it is submitted with *only* a digest and no tag.

I'm pretty sure this is what we want and should be considered correct behavior?

I don't want it to *not* be set and assume.

Reply:
Ok, I think that should plumb an empty string all the way through then which should be fine.